tools/c7n_azure/c7n_azure/resources/vm.py

```diff
@@ -23,4 +24,21 @@ class resource_type(object):
         client = 'ComputeManagementClient'
         enum_spec = ('virtual_machines', 'list_all')
         id = 'id'
-        name = 'name'
\ No newline at end of file
+        name = 'name'
+        default_report_fields = (
+            'name',
+            'location',
+            'properties.hardwareProfile.vmSize',
+        )
+
+@VirtualMachine.filter_registry.register('instance-view')
+class InstanceViewFilter(ValueFilter):
+    schema = type_schema('instance-view', rinherit=ValueFilter.schema)
+
+    def __call__(self, i):
+        if 'instanceView' not in i:
+            client = self.manager.get_client()
+            instance = client.virtual_machines.get(i['id'].split('/')[4], i['name'], expand='instanceview').instance_view
```

Comment:
might be worth getting a utility function for this

Reply:
👍 Yeah @anastasiia-zolochevska was actually thinking about overriding `augment` for ARM resources to set a `resourceGroup` field.  Sort of the most common field in the whole resource, will be good to have it readily accessible.  (or put some utility method in and a custom filter)

Reply:
Updated to use resourcegroup that has since been augmented into all ARM objects